dagmc_stats/DagmcQuery.py

```diff
@@ -236,3 +303,268 @@ def calc_area_triangle(self, tris=[]):
+                d_i = np.arccos((side_length_sum_sq_half -
+                                (side_i**2)) * side_i / side_length_prod)
+                bar = np.array((tri, vert_i, d_i, side_i),
+                               dtype=tri_vert_struct)
+                tri_vert_data[tri_vert_index] = bar
+                tri_vert_index += 1
+        return tri_vert_data, list(all_verts)
+
+    def __calc_gaussian_curvature(self, tri_vert_data):
+        """Get gaussian curvature values of all non-isolated vertices
+
+        inputs
+        ------
+        tri_vert_data : numpy structured array that stores the triangle and
+        vertex related data
+
+        outputs
+        -------
+        gc_all : dictionary in the form of vertex : gaussian curvature value
+        of the vertex
+        """
+        verts = self.get_verts()
+        gc_all = {}
+        for vert_i in verts:
+            gc_all[vert_i] = self.__gaussian_curvature(vert_i, tri_vert_data)
+        return gc_all
+
+    def __gaussian_curvature(self, vert_i, tri_vert_data):
+        """Get gaussian curvature value of a vertex
+        Reference: https://www.sciencedirect.com/science/article/pii/
+        S0097849312001203
+        Formula 1
+
+        inputs
+        ------
+            vert_i : vertex entity handle
+            tri_vert_data : numpy structured array that stores the
+            triangle and vertex related data
+
+        outputs
+        -------
+            gc : gaussian curvature value of the vertex
+        """
+        vert_entries = tri_vert_data[tri_vert_data['vert'] == vert_i]
+        sum_alpha_angles = sum(vert_entries['angle'])
+        gc = np.abs(2 * np.pi - sum_alpha_angles)
+        return gc
+
+    def __get_lri(self, vert_i, gc_all, tri_vert_data):
+        """Get local roughness value of a vertex
+        Reference: https://www.sciencedirect.com/science/article/pii/
+        S0097849312001203
+        Formula 2, 3
+
+        inputs
+        ------
+            vert_i : vertex entity handle
+            gc_all : dictionary in the form of vertex : gaussian curvature
+                value of the vertex
+            tri_vert_data : numpy structured array that stores the triangle
+                and vertex related data
+
+        outputs
+        -------
+            Lri : local roughness value of the vertex
+        """
+        DIJgc_sum = 0
+        Dii_sum = 0
+        all_tris = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert_i, 2, op_type=0)
+        meshset_tris = self.get_tris()
+        adj_tris = list(set(all_tris) & set(meshset_tris))
+        vert_j_list = list(self.dagmc_file._my_moab_core.get_adjacencies(
+            adj_tris, 0, op_type=1))
+        vert_j_list.remove(vert_i)
+        for vert_j in vert_j_list:
+            # get tri_ij_list (the list of the two triangles connected to both
+            # vert_i and vert_j)
+            tri_i_list = self.dagmc_file._my_moab_core.get_adjacencies(
```

Comment:
Don't we have this already in `all_tris`?

Reply:
also, if `vert_i` is connected to `vert_j` along the edge of the meshset, do we still use both triangles in this calculation?  one of them is outside of the meshset.

Reply:
no we don't use both in this context.